plugins/inputs/proxmox/structs.go

```diff
@@ -0,0 +1,51 @@
+package proxmox
+
+import (
+	"encoding/json"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"net/url"
+)
+
+type Proxmox struct {
+	BaseUrl         string            `toml:"base_url"`
+	ApiToken        string            `toml:"api_token"`
+	ResponseTimeout internal.Duration `toml:"response_timeout"`
```

Comment:
Optional, we have config.Duration now which wraps the time.Duration instead of composing it. Fine either way.